src/components/Main/Schedule/services.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  DayIndexOptions,
+  MappedScheduleInterface,
+  WeeklyEventData,
+  Divider,
+  WeeklyAssignmentInterface,
+  EventData,
+  DutyData,
+  TaskData,
+} from '../../../shared/types';
+import { SCHEDULE } from './database';
+
+const idxToMonth = [
+  'Jan',
+  'Feb',
+  'Mar',
+  'Apr',
+  'May',
+  'June',
+  'July',
+  'Aug',
+  'Sept',
+  'Oct',
+  'Nov',
+  'Dec',
+];
+
+const dayIndex: DayIndexOptions = {
+  Sunday: 0,
+  Monday: 1,
+  Tuesday: 2,
+  Wednesday: 3,
+  Thursday: 4,
+  Friday: 5,
+  Saturday: 6,
+};
+
+const readableDate = (unreadableDate: Date) =>
+  `${
+    unreadableDate.getMonth() + 1
+  }/${unreadableDate.getDate()}/${unreadableDate.getFullYear()}`;
+
+const incrementDay = (date: Date) => new Date(date.setDate(date.getDate() + 1));
+
+function determineStartDate(startDate: string, day: number) {
+  let current = incrementDay(new Date(startDate));
+  while (current.getDay() !== day) current = incrementDay(current);
+  return current;
+}
+
+export function everyRepeatingDayBetweenTwoDates(
+  startDate: string,
+  endDate: string,
+  day: string,
+) {
+  let everyRepeatingDay = [];
+  let start = new Date(startDate);
+
+  if (start.getDay() !== dayIndex[day])
+    start = determineStartDate(startDate, dayIndex[day]);
+
+  let current = new Date(start);
+  let end = new Date(endDate);
+
+  while (current <= end) {
+    everyRepeatingDay.push(readableDate(current));
+    current = new Date(current.setDate(current.getDate() + 7));
+  }
+  return everyRepeatingDay;
+}
+
+const zeroingDates = (num1: number, num2: number): string => {
+  num1++;
+  const month = num1.toString().length > 1 ? num1.toString() : `0${num1.toString()}`;
+  const day = num2.toString().length > 1 ? num2.toString() : `0${num2.toString()}`;
+  return `${month}/${day}`;
+};
```

Comment:
I think you mean padding. Also, no need to call `toString` 6 times:
```suggestion
const zeroPaddingDates = (monthIndex: number, date: number): string => {
  let MM = (monthIndex+1).toString();
  MM = MM.length > 1 ?  MM: `0${MM}`;
  let dd = day.toString();
  dd = dd.length > 1 ? dd : `0${dd}`;
  return `${MM}/${dd}`;
};
```
and then update `columnizedDates` and `contrivedDate`

Reply:
ahh much better thanks

Reply:
If you'd like to use the suggestion, GitHub should show the "Commit suggestion" button on your screen. Or I can click them for you next time. lemme know